spec/inputs/file_spec.rb

```diff
@@ -199,63 +184,101 @@
     end
   end
 
-  context "when #run is called multiple times", :unix => true do
+  context "when close_older config is specified" do
     let(:tmpdir_path)  { Stud::Temporary.directory }
     let(:sincedb_path) { Stud::Temporary.pathname }
-    let(:file_path)    { "#{tmpdir_path}/a.log" }
-    let(:buffer)       { [] }
-    let(:lsof)         { [] }
-    let(:stop_proc) do
-      lambda do |input, arr|
-        Thread.new(input, arr) do |i, a|
-          sleep 0.5
-          a << `lsof -p #{Process.pid} | grep "a.log"`
-          i.stop
-        end
+    let(:conf)         { Hash.new }
+    let(:events)       { [] }
+    let(:codec)        { CodecTracer.new }
+    let(:line)         { "line1.1-of-a" }
+
+    subject { described_class.new(conf) }
+
+    before do
+      conf.update(
+            "type" => "blah",
+            "path" => "#{tmpdir_path}/*.log",
+            "sincedb_path" => sincedb_path,
+            "stat_interval" => 0.02,
+            "codec" => codec,
+            "close_older" => 2,
+            "delimiter" => FILE_DELIMITER)
+
+      subject.register
+      Thread.new { subject.run(events) }
+    end
+
+    it "having timed_out, the identity is evicted" do
+      sleep 0.1
+      File.open("#{tmpdir_path}/a.log", "a") do |fd|
+        fd.puts(line)
       end
+      sleep 1
```

Comment:
this thread is to make sure the data is written to disk? We could probably use `File#fsync` here?